File: topicos/imperativa/lista1/go/ex1-even-number-list.go
package main

import "fmt"

func main() {
	var input int
	var counter int

	fmt.Printf("Digite o tamanho da lista: ")
	fmt.Scan(&input)

	list := make([]int, input)

	fmt.Printf("Entre com uma lista de numeros separadas por espaco e digite ENTER quando terminar:")

	for i := 0; i < input; i++ {
		fmt.Scan(&list[i])
	}

	for _, item := range list {
		if item%2 == 0 {
			counter++
		}
	}

	fmt.Printf("vc digitou %d numeros pares", counter)
}

File: topicos/imperativa/node/ex1-even-number-list.js
const readline = require('readline/promises');
const { stdin: input, stdout: output } = require("process");

async function readAndPrint() {
  const rl = readline.createInterface({
    input,
    output,
  });

  const strList = await rl.question(' Entre com uma lista separada por espaco e pressione ENTER quando: ');
  
  console.log(strList);

  const list = strList
      .split(" ")
      .map(i => Number(i))
      .filter(i => i%2 == 0);

  console.log(`${list.length} numeros pares foram inseridos;`);

  rl.close();
}

readAndPrint();


